### PR TITLE
[API] Allow listing kind-specific functions

### PR DESCRIPTION
**Pull request by liranbg**
It is required to let clients filter functions by their kind (remote / local / handler / ... all other runtime kinds)
this way, clients such as sdk / ui can show better granular tables filtering by the function kinds

https://iguazio.atlassian.net/browse/ML-7075

### CODE REVIEW

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -1389,7 +1389,9 @@ def list_functions(
         name: Optional[str] = None,
         project: Optional[str] = None,
         tag: Optional[str] = None,
+        kind: Optional[str] = None,
```
Comment (alonmr) [inline mlrun/db/httpdb.py]: I believe we need to be adding parameters at the end of functions to keep BC of clients using positional arguments

File: server/py/framework/db/sqldb/db.py [hunk 2]
```diff
@@ -1898,6 +1898,7 @@ def store_function(
         fn.updated = updated
         labels = get_in(function, "metadata.labels", {})
         update_labels(fn, labels)
+        fn.kind = function.pop("kind", None)
```
Comment (alonmr) [inline server/py/framework/db/sqldb/db.py:1903]: IIUC you pop to avoid data duplication. I think it's worth adding a comment

**Review by alonmr — COMMENTED** [review]
Looks good! Minor comment

**Review by alonmr — APPROVED** [review]

File: mlrun/db/base.py [hunk 3]
```diff
@@ -213,11 +213,13 @@ def delete_function(self, name: str, project: str = ""):
     def list_functions(
         self,
         name: Optional[str] = None,
-        project: Optional[str] = "",
-        tag: Optional[str] = "",
+        project: Optional[str] = None,
+        tag: Optional[str] = None,
+        kind: Optional[str] = None,
```
Comment (moranbental) [inline mlrun/db/base.py:218]: Move the `kind` and `format_` fields to the bottom

File: mlrun/db/nopdb.py [hunk 4]
```diff
@@ -230,12 +230,14 @@ def delete_function(self, name: str, project: str = ""):
...
+        name: Optional[str] = None,
+        project: Optional[str] = None,
+        tag: Optional[str] = None,
+        kind: Optional[str] = None,
         labels: Optional[Union[str, dict[str, Optional[str]], list[str]]] = None,
-        since=None,
-        until=None,
+        format_: mlrun.common.formatters.FunctionFormat = mlrun.common.formatters.FunctionFormat.full,
+        since: Optional[datetime.datetime] = None,
+        until: Optional[datetime.datetime] = None,
```
Comment (moranbental) [inline mlrun/db/nopdb.py:240]: Move the `kind` and `format_` fields to the bottom

File: server/py/framework/db/sqldb/db.py [hunk 5]
```diff
@@ -4807,6 +4816,9 @@ def _find_functions(
         if hash_key is not None:
             query = query.filter(Function.uid == hash_key)
 
+        if kind is not None:
```
Comment (moranbental) [inline server/py/framework/db/sqldb/db.py:4819]: if kind is an empty string?

**Review by moranbental — CHANGES_REQUESTED** [review]
Looks good! just minor suggestion

File: server/py/framework/db/sqldb/db.py (hunk 5 above)
Comment (liranbg) [reply server/py/framework/db/sqldb/db.py:4819]: there are function with kind empty (used to be local or handler, cant really recall) - and that should take action

**Pull request merged**